datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -19,45 +21,76 @@
     private final BookRepository repository;
     private final BookModelAssembler assembler;
     private final BookStoreRepository storeRepository;
+    private ServerMap map;
 
-    public BookController(BookRepository repository, BookModelAssembler assembler, BookStoreRepository storeRepository) {
+    public BookController(BookRepository repository, BookModelAssembler assembler, BookStoreRepository storeRepository, ServerMap map) {
         this.repository = repository;
         this.assembler = assembler;
         this.storeRepository = storeRepository;
+        this.map = map;
     }
 
     @PostMapping("/bookstores/{storeID}/books")
-    protected ResponseEntity<EntityModel<Book>> newBook(@RequestBody Book book, @PathVariable Long storeID){
-        BookStore store = storeRepository.getById(storeID);
-        book.setStoreID(storeID);
-        book.setStore(store);
-        EntityModel<Book> entityModel = assembler.toModel(repository.save(book));
-
-        return ResponseEntity
+    protected ResponseEntity newBook(@RequestBody Book book, @PathVariable Long storeID) throws Exception{
```

Comment:
Can't we still type the generic type?

`ResponseEntity<EntityModel<Book>>` did you have trouble with that?

Reply:
We need both so we can return either the EntityModel or the redirect. Therefore we need a Generic ResponseEntity that can be either

Reply:
yeah, I am wondering if it will still work? Do you get a compilation error?